### PR TITLE
feat: Add session viewer

### LINKED ISSUE #847: Add support for multiple parallel sessions with different tools
It would be helpful to run multiple sessions with different tools in parallel. For example, it would make sense to run Capella & Jupyter in parallel. 

On the other hand, we need a way to specify which tools are working together and which not. For example, two Eclipse based tools won't run with the same workspace.

### PR DESCRIPTION
**Pull request by MoritzWeber0**
This PR adds a session viewer into the Collaboration Manager. With the session overview, you can open multiple sessions within one browser tab and in the Collaboration Manager directly. It provides many options for extendability, e.g. session tokens and the workspace file management could be opened directly from the view in the future.

The video is commented, so don't forget to turn on the sound :)

https://github.com/DSD-DBS/capella-collab-manager/assets/23395732/df3b9e0f-5052-4392-8808-515caeb64e3b

Another feature is automatic resizing (video without sound):

https://github.com/DSD-DBS/capella-collab-manager/assets/23395732/9af1d544-7406-420d-a153-6603c8115e6f

Is a fundamental feature for #848.
Depends on https://github.com/DSD-DBS/capella-dockerimages/pull/172

TODO: 

- [x] Restrict sessions which can run in parallel
- [x] Enable clipboard & keyboard for Guacamole sessions in `iframe`
- [x] Add handling if no session exists
- [x] Add handling if session is not in running state

To start a development environment, you have to add this section to `config.yaml`:
```
general:
  host: localhost
  port: "4200"
  scheme: http
```

### CODE REVIEW

File: frontend/src/styles.css [hunk 1]
```diff
@@ -13,6 +13,11 @@ html,
 body {
   height: 100%;
 }
+
+html {
+  overflow-y: scroll;
+}
+
```
Comment (amolenaar) [inline frontend/src/styles.css:20]: We should not have to define this, right? I mean: scroll on overflow is the default.

File: backend/capellacollab/sessions/operators/k8s.py [hunk 2]
```diff
@@ -177,6 +179,7 @@ def start_persistent_jupyter_session(
             "JUPYTER_BASE_URL": path,
             "JUPYTER_TOKEN": token,
             "JUPYTER_PORT": "8888",
+            "HOST": f"{general_conf.get('scheme')}://{general_conf.get('host')}:{general_conf.get('port')}",
```
Comment (amolenaar) [inline backend/capellacollab/sessions/operators/k8s.py:182]: I would use a different name than `HOST`.

Normally the `HOST` environment variable contains the machine/container host name (no scheme and port).

File: frontend/src/app/sessions/session/session.component.html [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      }}<span *ngIf="session.type === 'readonly'">
+        (project {{ session.project!.name }})</span
+      >, created at {{ session.created_at | date: "EE, dd MMM y HH:mm:ss" }}
+    </mat-checkbox>
+  </div>
+  <div class="border-2 p-2 mt-2 rounded w-fit">
+    Please note: We're not technically preventing that you select two Guacamole
+    sessions (for example two Capella sessions), <br />but the Guacamole server
+    will not let you connect to the sessions. We're working on making this
+    possible.
```
Comment (amolenaar) [inline frontend/src/app/sessions/session/session.component.html:47]: I would phrase this differently. Two sessions using Eclipse based tooling.

It's completely transparent to the users that we use Guacamole to expose the sessions.

**Review by amolenaar — CHANGES_REQUESTED** [review]
When I try this in Firefox, both sessions remain "Not focused" and I cannot do anything.

Currently the "Session viewer" is in the top row. I think it should be part of the "Sessions" page. "Session viewer" is quite a generic name and from the title of the button you cannot really derive what it does or how it's different from "Sessions".

File: backend/capellacollab/sessions/routes.py [hunk 4]
```diff
@@ -280,24 +280,46 @@ def request_persistent_session(
...
+            for session in existing_user_sessions
+            if session.tool.name != "Jupyter"
+        ]:
+            raise fastapi.HTTPException(
+                status_code=status.HTTP_409_CONFLICT,
+                detail={
+                    "err_code": "EXISTING_SESSION",
+                    "reason": (
+                        "You already have a open persistent Eclipse-based session.",
+                        "Currently, we can only run Eclipse-based session at a time.",
```
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply backend/capellacollab/sessions/routes.py:318]: Reason can be a list or string. If it's a list, it is converted to a string in the frontend.

File: docs/user/docs/release-notes.md [hunk 5]
```diff
@@ -5,35 +5,35 @@
...
 
-# v2.10.0
+## v2.10.0
 
 Adds a model complexity badge integration. Once set up, a model complexity
 badge can be seen in the model overview. More information
 [here](./projects/models/complexity_badge.md).
 
-# v2.9.0
+## v2.9.0
```
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply docs/user/docs/release-notes.md:36]: The file will be removed in https://github.com/DSD-DBS/capella-collab-manager/pull/874 anyway :)

File: frontend/src/app/app.component.css [hunk 6]
```diff
@@ -7,6 +7,12 @@
   display: flex;
   flex-direction: column;
   min-height: 100%;
+  overflow: unset;
+}
+
+.container-wrapper {
+  overflow-y: visible;
+  overflow: unset;
```
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply frontend/src/app/app.component.css:15]: Unfortunately yes. I had two scroll bars with the default style.

File: frontend/src/styles.css (hunk 1 above)
Comment (MoritzWeber0) [reply frontend/src/styles.css:20]: The default is `visible`, not `scroll`.

File: docs/user/docs/release-notes.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply docs/user/docs/release-notes.md]: Rebased onto `release-notes`, the changes are removed.

File: backend/capellacollab/sessions/routes.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply backend/capellacollab/sessions/routes.py:318]: Good catch! :)

File: backend/capellacollab/sessions/operators/k8s.py (hunk 2 above)
Comment (MoritzWeber0) [reply backend/capellacollab/sessions/operators/k8s.py:182]: What about `ORIGIN_HOST`? It has to be changed in the Jupyter image of capella-dockerimages repository too.

File: frontend/src/app/sessions/session/session.component.html
Comment (MoritzWeber0) [inline frontend/src/app/sessions/session/session.component.html:1]: > When I try this in Firefox, both sessions remain "Not focused" and I cannot do anything.

Comment (MoritzWeber0) [reply frontend/src/app/sessions/session/session.component.html:1]: We should look into it together. Unfortunately, DB forbids the installation for Firefox, so I'm not able to try it out.

Comment (MoritzWeber0) [inline frontend/src/app/sessions/session/session.component.html:1]: > Currently the "Session viewer" is in the top row. I think it should be part of the "Sessions" page. "Session viewer" is quite a generic name and from the title of the button you cannot really derive what it does or how it's different from "Sessions".

To not overload the main page, maybe just add a button: "Open internal session viewer >" below the list of active sessions?

Comment (MoritzWeber0) [reply frontend/src/app/sessions/session/session.component.html:1]: To not overload the main page, maybe just add a button: "Open internal session viewer >" below the list of active sessions?

File: frontend/src/app/app.component.css (hunk 6 above)
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply frontend/src/app/app.component.css:15]: The proposed change doesn't seem to work on Chrome.

[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply frontend/src/app/app.component.css:15]: I think the reason is that `overflow-y` overwrites `overflow`. And `overflow` is set in the default style.

[the remark this answers is not in the document]
Comment (amolenaar) [reply frontend/src/app/app.component.css:15]: Ah, it has to do with how the styles are compiled in the components. Never mind, then.

File: frontend/src/styles.css (hunk 1 above)
Comment (amolenaar) [reply frontend/src/styles.css:20]: ok

File: frontend/src/app/sessions/session/session.component.html
Comment (amolenaar) [reply frontend/src/app/sessions/session/session.component.html:1]: I think that is an improvement.

File: backend/capellacollab/sessions/operators/k8s.py (hunk 2 above)
Comment (amolenaar) [reply backend/capellacollab/sessions/operators/k8s.py:182]: `ORIGIN_HOST` sounds fine. Or something like `CSRF_ORIGIN_HOST`?

Comment (MoritzWeber0) [reply backend/capellacollab/sessions/operators/k8s.py:182]: Renamed to `CSRF_ORIGIN_HOST`.

**Pull request merged**